### PR TITLE
feat: amount / currency input with live formatting

### LINKED ISSUE #3176: User settings should be taken into account when validating and sending inputs for values
As brought up in #3175 , specifically [this](https://github.com/rotki/rotki/pull/3175#issuecomment-879050431) comment, we should utilize the user setting for decimal (and potentially thousands) separators to:
(a) determine which decimal separator is acceptable for value inputs (e.g. form validation), and 
(b) whether we need to transform the input before sending it to the backend or not. (e.g. if user has thousands separator `.` and decimal separator `,` a valid value input would be "1.354,42" and then transformed to "1354.42" which is what the backend expects)

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #3176 

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.
- [ ] Create `<amount-input>` component that provide formatting while input (improve UX).
- [ ] Add a restriction for `thousandSeparator` and `decimalSeparator` setting, where both of them can't have same value, cause it will break the amount formatting while input.
- [ ] Use `<amount-input>` to replace all input that behave as amount / currency field. I maybe haven't replace all usage, so it should be done along the development.

### CODE REVIEW

**Comment by kelsos** [comment]
Since we are using `npm ci` and we depend on the workspace's lock file, when adding new dependencies it needs to be properly updated.

To do so, when installing packages you can run `npm install package --save-exact -w rotki` from the workspace root `frontend/` for runtime dependencies and you can add -D for development dependencies. `-w rotki` will also update the `frontend/app/package.json` 

To test the PR I took care of it and pushed also the changes.

Another thing I noticed I missed completely in the previous PRs, was adding changelog entries.

We are usually adding the changelog entries in the PRs. As a rule of thumb bug fixes _(bug fixes for develop only bugs should be excluded)_ and features should have a new changelog entry. 

I added the changelog changes for the previously merged PRs.

I checked the PR it seems to work fine but I noticed the following:

image

A rebase might also be needed.

**Comment by lukicenturi** [comment]
> Member

This should be updated now @kelsos

File: frontend/app/src/components/ExternalTradeForm.vue [hunk 1]
```diff
@@ -540,13 +540,17 @@ export default class ExternalTradeForm extends Vue {
       this.selectedCalculationInput = 'quoteAmount';
       this.$nextTick(() => {
         const quoteAmountInput = this.$refs.quoteAmountInput as any;
-        if (quoteAmountInput) quoteAmountInput.focus();
+        if (quoteAmountInput) {
+          quoteAmountInput.$el.querySelector('input').focus();
```
Comment (kelsos) [inline frontend/app/src/components/ExternalTradeForm.vue:544]: Wouldn't it be a bit better to add a `focus` method to `AmountInput` and have it handle this? the $el is the v-text-field that is inside AmountInput, correct?

**Comment by kelsos** [comment]
I also see that my additions are not in the PR anymore, they were overwritten

File: frontend/app/src/components/ExternalTradeForm.vue (hunk 1 above)
Comment (lukicenturi) [reply frontend/app/src/components/ExternalTradeForm.vue]: Updated @kelsos, you mean like this?

**Review by kelsos — APPROVED** [review]
lgtm

**Pull request merged**